src/foam/u2/view/RichChoiceView.js

```diff
@@ -321,9 +326,17 @@ foam.CLASS({
       documentation: 'The text that the user typed in to search by.',
       postSet: function(oldValue, newValue) {
         this.sections.forEach((section) => {
-          section.filteredDAO = newValue
-            ? section.dao.where(this.KEYWORD(newValue))
-            : section.dao;
+          if ( newValue ) {
+            var pred = this.KEYWORD(newValue);
```

Comment:
Could set 'pred' in the 'else' of the below 'if' rather than create a KEYWORD which gets thrown out. Would also make the code easier to understand.